overlord/hookstate/hookmgr.go

```diff
@@ -295,6 +318,19 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 		return nil, err
 	}
 
+	// add timeout handling
+	var killTimer *time.Timer
+	if maxRuntime > 0 {
+		killTimer = time.AfterFunc(maxRuntime, func() {
```

Comment:
thanks @pedronis - I changed to a timer channel as suggested. I would still like to keep the messages for abort and timeout different as things are different. Unfortuantely this leads to code duplications, ideas welcome.